web/src/drafts.js

```diff
@@ -360,9 +360,14 @@ export function format_draft(draft) {
                     draft_model.editDraft(id, draft);
                 }
             }
+        } else {
+            const sub = stream_data.get_sub(stream_name);
+            if (sub !== undefined) {
+                draft.stream_id = sub.stream_id;
+            }
```

Comment:
Yeah it's new logic because we now need to make sure there's a `stream_id` set, even if we only have the stream name (unless it definitely doesn't have an id).

Reply:
We can't rely on the compose state for drafts -- this formats all drafts, not just ones for a possibly open composebox.